common/plugins/eu.esdihumboldt.hale.common.instance/src/eu/esdihumboldt/hale/common/instance/model/impl/MultiInstanceCollection.java

```diff
@@ -194,6 +194,7 @@ public Instance next() {
 						currentIterator.close();
 					currentCollection = hasNextIndex;
 					currentIterator = collections.get(currentCollection).iterator();
+					currentIterator.hasNext(); // force to initialize currentIterator.hasNext flag 
```

Comment:
What difference does it make?
My feeling is that if this is necessary we should try to fix the cause and not the symptom.